src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/DataGridCell.cs

```diff
@@ -966,43 +972,44 @@ protected override void OnKeyDown(KeyEventArgs e)
         {
             if (!e.Handled)
             {
-                const ModifierKeys ModifierMask = ModifierKeys.Alt | ModifierKeys.Control | ModifierKeys.Shift | ModifierKeys.Windows;
-                ModifierKeys modifierKeys = Keyboard.Modifiers & ModifierMask;
-
-                if (((e.SystemKey == Key.Right) || (e.SystemKey == Key.Left)) && (modifierKeys == ModifierKeys.Alt))
+                if (!OptOutOfGridColumnResizeUsingKeyboard)
                 {
-                    DataGridLength updatedWidth = new DataGridLength();
-
-                    if (e.SystemKey == Key.Right)
+                    ModifierKeys modifierKeys = Keyboard.Modifiers & ModifierMask;
+                    if (((e.SystemKey == Key.Right) || (e.SystemKey == Key.Left)) && (modifierKeys == ModifierKeys.Alt))
                     {
-                        updatedWidth = new DataGridLength(Column.ActualWidth + ColumnWidthStepSize);
-                    }
-                    else if (e.SystemKey == Key.Left)
-                    {
-                        updatedWidth = new DataGridLength(Column.ActualWidth - ColumnWidthStepSize);
-                    }
+                        DataGridLength updatedWidth = new DataGridLength();
 
-                    if (Column != null)
-                    {
-                        if (Column.CanColumnResize(updatedWidth))
+                        if (e.SystemKey == Key.Right)
                         {
-                            Column.SetCurrentValueInternal(DataGridColumn.WidthProperty, updatedWidth);
+                            updatedWidth = new DataGridLength(Column.ActualWidth + ColumnWidthStepSize);
                         }
-                        e.Handled = true;
+                        else if (e.SystemKey == Key.Left)
+                        {
+                            updatedWidth = new DataGridLength(Column.ActualWidth - ColumnWidthStepSize);
+                        }
+
+                        if (Column != null)
```

Comment:
I don't understand why Column can be null here. If you see line 1005:
```c#
if ((e.Key == Key.F3) && Column.CanUserSort)
```
where it is not checked. Either the column can never be null here, or line 1005 we need to add a null check as well?

Reply:
Added null check at line 1005